### DISCUSSION TITLE
Módulo: Cronômetro (dúvida)

### DISCUSSION THREAD

**Discussion by ghost** [balta-io/2801#77] [Perguntas e Respostas] [answered]
Olá!
Fiquei na dúvida em como fazer o comando para sair no código. Por exemplo, o Balta colocou a opção no menu de quando o usuário digitasse 0 sairia da aplicação, só que quando coloca 0 dá erro. Tentei fazer do jeito que foi o primeiro mão na massa da calculadora utilizando switch, case... e até dá certo mas quando coloco 10s ou qualquer outro número com "s" ou "m" dá erro. A função mesmo do cronômetro não está imprimindo. Alguém poderia me ajudar?

O erro:

`Unhandled exception. System.FormatException: Input string was not in a correct format. 
   at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
   at System.Int16.Parse(String s)
   at Stopwatch.Program.Menu() in C:\dev\Stopwatch\Program.cs:line 21
   at Stopwatch.Program.Main(String[] args) in C:\dev\Stopwatch\Program.cs:line 10 `

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @mayteferre, como vai?

Consegue colocar a parte do código que dá erro aqui? Assim posso analisar e te ajudar melhor!

**Reply by ghost** [reply]
Oiee!! 
Na verdade, vou refazer minha pergunta por que revi o vídeo e vi que você chegou a fazer um if para sair da aplicação só que no final dá erro.  Gostaria de saber como resolver esse erro...
Agradeço desde já :)


![image](https://user-images.githubusercontent.com/96352742/168336306-28e23daf-924a-4601-a22f-6b5b1a26c9de.png)

**Reply by ghost** [reply]
Olá. Provavelmente o Balta vai conseguir te ajudar somente tendo acesso a este print, mas para tirar dúvidas eu sugiro compartilhar o repositório inteirinho da aplicação, juntamente com a descrição da tua dúvida.

**Reply by ghost** [reply]
Então, no próprio github tem o código -> https://github.com/balta-io/2801/blob/master/Modulo%2008%20-%20Stopwatch/Program.cs
Frisei essa parte no print por que é onde está minha dúvida em relação a aplicação do `if` e o `System.Enviroment.Exit(0);`, pois ao dar dotnet run e digitar 0 ele dá erro. Acredito que seja por causa do "-1" acima, no `data.Length`, mas não sei como resolver :(

**Reply by ghost** [reply]
Oi @LazieWouters , agradeço demais sua explicação em relação ao `Substring(data.Length - 1, 1)`, muito esclarecedora!  Mas a minha dúvida/curiosidade é no que se refere ao digitar somente o "0" e o `System.Enviroment.Exit(0);` funcionar. Qual solução você sugere pra esse problema?

**Reply by andrebaltieri** [reply]
@mayteferre o @LazieWouters respondeu a questão.
Em relação ao `System.Environment.Exit(0);`, é uma outra função, que espera um inteiro como parâmetro.

**Reply by j3mfilho** [reply]
@andrebaltieri Ela espera um inteiro, mas no caso ali o variável "time" está vazia, sendo assim ela não consegue fazer a comparação se é 0. Tentei comparar caso ela fosse nula, mas pelo jeito não é assim que funciona, pois o erro permanece.

**Reply by j3mfilho** [reply]
Só para colocar que resolvi fazendo uma condicional simples, como abaixo: 

if(data.Length == 1){
                      time = int.Parse(data.Substring(0,1)) ;
                }
                else{
                     time = int.Parse(data.Substring(0,data.Length-1)) ;
                }
